test/functional/lua/with_spec.lua

```diff
@@ -0,0 +1,296 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+local Screen = require('test.functional.ui.screen')
+
+local fn = n.fn
+local api = n.api
+local command = n.command
+local eq = t.eq
+local exec_lua = n.exec_lua
+local matches = t.matches
+local pcall_err = t.pcall_err
+
+before_each(function()
+  n.clear()
+end)
+
+describe('vim._with {win = }', function()
```

Comment:
I copy pasted the tests for `nvim_buf_call` and `nvim_win_call` and used `vim._with` instead. It's a bit code duplication but it's fine since it's for tests IMO. This means testing for the contexts for `buf`, `win` and `options` are pretty much covered. I threw in a test for `:lockmarks` since it was the original justification for this apparently. I haven't added tests for the 500 other context options but I figure we can work on this in iterations.

Reply:
I noticed :) I think this is the right decision. Hopefully we can deprecate `nvim_buf_call` and `nvim_win_call` soon...